wallet-new/src/Cardano/Wallet/API/V1/Migration.hs

```diff
@@ -0,0 +1,105 @@
+    , lowerV1Monad
+    , Migrate(..)
+    -- * Isomorphisms
+    , walletAssurance
+    -- * Configuration re-exports
+    , HasInfraConfiguration
+    , HasSscConfiguration
+    , HasCompileInfo
+    , HasConfiguration
+    , HasUpdateConfiguration
+    , HasNodeConfiguration
+    ) where
+
+import           Universum
+
+import qualified Cardano.Wallet.API.V1.Types      as V1
+import qualified Pos.Wallet.Web.ClientTypes.Types as V0
+
+import           Control.Lens
+import qualified Control.Monad.Catch              as Catch
+import           Mockable                         (runProduction)
+import           Servant
+
+import           Pos.Configuration                (HasNodeConfiguration)
+import           Pos.Core.Configuration           (HasConfiguration)
+import           Pos.Infra.Configuration          (HasInfraConfiguration)
+import           Pos.Ssc                          (HasSscConfiguration)
+import           Pos.Update.Configuration         (HasUpdateConfiguration)
+import           Pos.Util.CompileInfo             (HasCompileInfo)
+import           Pos.Wallet.Web.Mode              (WalletWebMode, WalletWebModeContext)
+
+-- | Temporary monad to handle the migration from the V0 & V1 stacks.
+type MonadV1   = WalletWebMode
+type V1Context = WalletWebModeContext
+
+-- | Hoist a 'V1' monad to a Servant's Handler.
+-- See: http://haskell-servant.readthedocs.io/en/stable/tutorial/Server.html#natural-transformations
+v1MonadNat :: V1Context -> (forall a. MonadV1 a -> Handler a)
+v1MonadNat = lowerV1Monad
+
+-- | Converts our domain-specific monad into a standard Servant `Handler`.
+lowerV1Monad :: V1Context -> MonadV1 a -> Handler a
+lowerV1Monad ctx handler =
+    liftIO (hoistHandler handler) `Catch.catches` excHandlers
+  where
+
+    hoistHandler :: forall a . MonadV1 a -> IO a
+    hoistHandler = runProduction . flip runReaderT ctx
+
+    excHandlers = [Catch.Handler throwError]
+
+-- | 'Migrate' encapsulates migration between types, when possible.
+class Migrate from to where
+    migrate :: from -> to
+
+--
+-- Instances
+--
+
+instance Migrate V0.CWallet V1.Wallet where
+    migrate V0.CWallet{..} =
+        V1.Wallet { V1.walId   = migrate @(V0.CId V0.Wal) cwId
+                  , V1.walName = V0.cwName cwMeta
+                  , V1.walBalance = migrate @V0.CCoin @V1.RenderedBalance cwAmount
+                  }
+
+--
+instance Migrate V0.CWalletAssurance V1.AssuranceLevel where
+    migrate V0.CWAStrict = V1.StrictAssurance
+    migrate V0.CWANormal = V1.NormalAssurance
+
+instance Migrate V1.AssuranceLevel V0.CWalletAssurance where
+    migrate V1.StrictAssurance = V0.CWAStrict
+    migrate V1.NormalAssurance = V0.CWANormal
+
+walletAssurance :: Iso' V0.CWalletAssurance V1.AssuranceLevel
+walletAssurance = iso migrate migrate
+
+--
+instance Migrate V0.CPassPhrase V1.SpendingPassword where
+  migrate (V0.CPassPhrase pp) = pp
+
+instance Migrate V1.SpendingPassword V0.CPassPhrase where
+  migrate pp = V0.CPassPhrase pp
```

Comment:
In some places indentation is 4 spaces, in some places it's 2 spaces. Can you stick to 4 spaces (as per style-guide) please?

Reply:
Is there a way to enforce this with stylish-haskell as well? I come from projects where we were using 2 spaces (which, btw, allow more test to be on one line, so that's surprising why you guys are using 4 spaces), so I will keep forgetting as muscle memory will kick in. 😢

Reply:
@adinapoli-iohk AFAIK, unfortunately, no, there's no way to control indentation with stylish-haskell. If you're using Emacs though, I can share my `.dir-locals.el` for this project, which sets up indentation accordingly to style guide.